src/client/src/rt-util/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react'
+
+export const useLocalStorage = <T>(key: string, initialState: T) => {
+    const [state, setState] = useState(() => {
+        return window.localStorage.getItem(key) ?? JSON.stringify(initialState)
+    })
+
+    useEffect(() => {
+        const updateState = () => {
+            setState(window.localStorage.getItem(key) ?? JSON.stringify(initialState))
+        }
+
```

Comment:
I think that it makes sense to invoke `updateState` right after its definition, so that if the `key` changes, then its value gets updated from `localStorage`

Reply:
Added.